include/souper/Inst/Inst.h

```diff
@@ -116,6 +116,9 @@ struct Inst : llvm::FoldingSetNode {
   static bool isAssociative(Kind K);
   static bool isCommutative(Kind K);
   static int getCost(Kind K);
+  unsigned KnownZeros;
+  unsigned KnownOnes;
+  std::string BitPattern;
```

Comment:
For orinting the bit pattern, I was just accessing the bit pattern from
instruction structure. I can actually call a function here to print the
pattern and this function would now be a member of structure. I will update
it.

On Thu, Jul 9, 2015 at 9:29 AM, John Regehr <notifications@github.com>
wrote:

> In include/souper/Inst/Inst.h
> <https://github.com/google/souper/pull/195#discussion_r34267793>:
>
> > @@ -116,6 +116,9 @@ struct Inst : llvm::FoldingSetNode {
> >    static bool isAssociative(Kind K);
> >    static bool isCommutative(Kind K);
> >    static int getCost(Kind K);
> > +  unsigned KnownZeros;
> > +  unsigned KnownOnes;
> > +  std::string BitPattern;
>
> is there a reason to add this string to Inst? why not just create it when
> needed?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/souper/pull/195/files#r34267793>.
>



-- 
Thanks and Regards,
Jubi Taneja